nb2/api.py

```diff
@@ -295,33 +428,120 @@ def __init__(self, *args, **kwargs):
-                    slack_user_id=slack_user_id
-                ),
+                message=self.ERRORS.get("person_does_not_exist").format(user_id=user_id),
             )
 
         if target_person.has_said(parsed_args.get("content")):
             return abort(
                 409,
-                message=self.ERRORS.get("already_exists").format(slack_user_id=slack_user_id),
+                message=self.ERRORS.get("already_exists").format(user_id=user_id),
             )
 
-        data = AddQuoteDTO(slack_user_id, content)
+        data = AddQuoteDTO(target_person, content)
         new_quote = add_quote_to_person(data)
 
         return marshal(new_quote, self.fields), 201
 
 
+class QuoteResource(QuoteResourceBase):
+    """
+    Implements the API methods for operating on singular Quotes.
+    """
+
+    ERRORS = {
+        "user_id_missing": "user_id is required",
+        "content_missing": "content is required",
+        "person_does_not_exist": (
+            "Can't add a quote to Person with user_id " "{user_id} because they don't exist."
+        ),
+        "quote_does_not_exist": (
+            "Can't find a quote with quote_id " "{quote_id} because it don't exist."
+        ),
+        "already_exists": (
+            "The Quote content provided can't be added because "
+            "it already exists for this Person."
+        ),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.parser.add_argument(
+            "user_id",
+            dest="user_id",
+            type=str,
+            help=self.ERRORS.get("user_id_missing"),
+        )
+        self.parser.add_argument(
+            "content",
+            dest="content",
+            type=str,
+            help=self.ERRORS.get("content_missing"),
+        )
+
+    def get(self, quote_id):
+        quote = get_quote(quote_id)
+
+        if quote is None:
+            abort(
+                404,
+                message=self.ERRORS["quote_does_not_exist"].format(quote_id=quote_id),
+            )
+        return marshal(quote, self.fields), 200
+
+    def patch(self, quote_id):
+        quote = get_quote(quote_id)
+        if quote is None:
+            abort(
+                404,
+                message=self.ERRORS["quote_does_not_exist"].format(quote_id=quote_id),
+            )
+
+        parsed_args = self.parser.parse_args()
+        user_id = parsed_args.get("user_id")
+        content = parsed_args.get("content")
+
+        if user_id:
+            target_person, is_active = get_person(user_id)
+        else:
+            target_person = get_person_by_quote(quote)
+
+        if not target_person:
+            return abort(
+                404,
+                message=self.ERRORS.get("person_does_not_exist").format(user_id=user_id),
+            )
+
+        if not content:
+            content = quote.content
+
+        if target_person.has_said(content):
```

Comment:
If someone didn't provide `content` in their body then they will always get the "this quote already exists" msg, should we change this to an `else if` and combine this with the `if` above (L523)? Or check that content is present first and move the first if after this